apm-sniffer/apm-sdk-plugin/jsonrpc4j-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jsonrpc4j/JsonRpcHttpClientInterceptor.java

```diff
@@ -0,0 +1,82 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.jsonrpc4j;
+
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import java.lang.reflect.Method;
+import java.net.URL;
+
+@SuppressWarnings("unused")
+public class JsonRpcHttpClientInterceptor implements InstanceMethodsAroundInterceptor, InstanceConstructorInterceptor {
+
+    @Override
+    public void onConstruct(EnhancedInstance objInst, Object[] objects) {
+        URL url = (URL) objects[1];
+        JsonRpcPeerInfo clientDto = new JsonRpcPeerInfo();
+        int port = url.getPort();
+        if (port < 0) {
+            if (isHttps(url)) {
+                port = 443;
+            } else {
+                port = 80;
+            }
+        }
+
+        clientDto.setPort(port);
+        clientDto.setServiceUrl(url);
+        objInst.setSkyWalkingDynamicField(clientDto);
+    }
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] objects, Class<?>[] classes, MethodInterceptResult result) throws Throwable {
+        JsonRpcPeerInfo clientDto = (JsonRpcPeerInfo) objInst.getSkyWalkingDynamicField();
+        String methodName = objects[0].toString();
+        String operationName = clientDto.getServiceUrl().getPath() + "." + methodName;
+        AbstractSpan span = ContextManager.createExitSpan(operationName, new ContextCarrier(), clientDto.getServiceUrl().getHost() + ":" + clientDto.getPort());
+        span.setComponent(ComponentsDefine.JSON_RPC);
+        Tags.HTTP.METHOD.set(span, "POST");
+        Tags.URL.set(span, clientDto.getServiceUrlString());
+        SpanLayer.asRPCFramework(span);
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] objects, Class<?>[] classes, Object o) throws Throwable {
+        ContextManager.stopSpan();
```

Comment:
Should we check the HTTP(s) response code to check the status? Rather than just simply exception?

Reply:
> Should we check the HTTP(s) response code to check the status? Rather than just simply exception?

The jsonrpc client  not check the response code, just get the response body, if other condition they use catch block get the error message. they not care about response code,  ref: https://github.com/briandilley/jsonrpc4j/blob/master/src/main/java/com/googlecode/jsonrpc4j/JsonRpcHttpClient.java#L120

Reply:
So, from user perspective, you can't read the response code even you want to?

Reply:
yeah, they use the error object define the error ,document ref:https://www.jsonrpc.org/specification#error_object